pyzim/pyzim.pyx

```diff
@@ -410,10 +214,26 @@ cdef class ZimCreator:
         new_metadata = {pascalize(key): value for key, value in kwargs.items()}
         self._metadata.update(new_metadata)
 
+    def _update_article_counter(self, ZimArticle article):
+        # default dict update
+        self._article_counter[article.mimetype] += 1
+
+    def add_article(self, ZimArticle article not None):
+        if not article.can_write:
+            raise RuntimeError("Article is not good for writing")
+
+        # Make a shared pointer to ZimArticleWrapper from the ZimArticle object (dereference internal c_article)
+        cdef shared_ptr[ZimArticleWrapper] art = make_shared[ZimArticleWrapper](dereference(article.c_article));
```

Comment:
I'm not sure about the dereference.
If the ZimArticleWrapper is a wrapper around a python article, we probably don't want to pass it a `*(article.c_article)`